frb_codegen/src/generator/dart/ty_delegate.rs

```diff
@@ -1,40 +1,58 @@
 use crate::generator::dart::gen_wire2api_simple_type_cast;
 use crate::generator::dart::ty::*;
 use crate::ir::*;
+use crate::target::Acc;
 use crate::type_dart_generator_struct;
-use crate::utils::BlockIndex;
 
 type_dart_generator_struct!(TypeDelegateGenerator, IrTypeDelegate);
 
 impl TypeDartGeneratorTrait for TypeDelegateGenerator<'_> {
-    fn api2wire_body(&self, block_index: BlockIndex) -> Option<String> {
-        Some(match self.ir {
-            IrTypeDelegate::String => {
-                "return _api2wire_uint_8_list(utf8.encoder.convert(raw));".to_string()
-            }
-            IrTypeDelegate::SyncReturnVecU8 => "/*unsupported*/".to_string(),
+    fn api2wire_body(&self) -> Acc<Option<String>> {
+        let wasm = self.context.config.wasm_enabled;
+        match self.ir {
+            IrTypeDelegate::String => Acc {
+                io: Some("return api2wire_uint_8_list(utf8.encoder.convert(raw));".into()),
+                wasm: wasm.then(|| "return raw;".into()),
```

Comment:
In this particular instance, I could see how creating a string doesn't make for much extra work and should be performed unconditionally. However there exists some branches of codegen (e.g. enums) that do quite a lot of work only to be discarded by non-WASM users. Nevertheless it's a minor issue.

Reply:
Well, I mean using a closure. e.g.

```
Acc {
  io: ...,
  wasm: || { some_code_here },
}
```

Just like what we do everyday: https://stackoverflow.com/questions/56726571/why-choosing-unwrap-or-else-over-unwrap-or

```
    let _one = None.unwrap_or("one".to_string()); // the existing method
    let _two = None.unwrap_or_else(|| "two".to_string()); // the method I propose

Reply:
I'm rethinking this and I don't think closures are a good long-term solution, due to the fact that `Acc<Option<String>>` is used as the return type of a *trait method*. Aside from the additional complexity, we would have to resort to dynamic dispatch aka `Acc<Option<&dyn Fn() -> String>>` which would cause more trouble than it solves. We can revisit this logic in the future, when we have a clearer picture of the common patterns between the impls.

Reply:
Sure. Shall we make a tracking issue as a backlog?